src/thread/pthread/SDL_systhread.c

```diff
@@ -286,7 +286,9 @@ SDL_SYS_SetThreadPriority(SDL_ThreadPriority priority)
 void
 SDL_SYS_WaitThread(SDL_Thread * thread)
 {
-    pthread_join(thread->handle, 0);
+    if (thread->state.value != SDL_THREAD_STATE_ZOMBIE) {
```

Comment:
The thread state value is set with SDL_AtomicCAS() and should be read atomically with SDL_AtomicGet()

```suggestion
    if (SDL_AtomicGet(&thread->state.value) != SDL_THREAD_STATE_ZOMBIE) {
```